src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol

```diff
@@ -0,0 +1,303 @@
+            address[] memory compliantNominees;
+            if (blacklistedNomineeCount_ > 0) {
+                compliantNominees = SecurityCouncilMgmtUtils.filterAddressesWithBlacklist(maybeCompliantNominees, blacklisted[proposalId], blacklistedNomineeCount_);
+            }
+            else {
+                compliantNominees = maybeCompliantNominees;
+            }
+
+            nominees = SecurityCouncilMgmtUtils.randomAddToSet(currentMembers, compliantNominees, targetNomineeCount, uint256(blockhash(block.number - 1)));
+        }
+        else if (blacklistedNomineeCount_ > 0) {
+            // there are exactly the right number of compliant nominees
+            // but some of the nominees have been blacklisted
+            // we should remove the blacklisted nominees from SecurityCouncilNomineeElectionGovernorCounting's list
+            address[] memory maybeCompliantNominees = SecurityCouncilNomineeElectionGovernorCountingUpgradeable.nominees(proposalId);
+            nominees = SecurityCouncilMgmtUtils.filterAddressesWithBlacklist(maybeCompliantNominees, blacklisted[proposalId], blacklistedNomineeCount_);
+        }
+        else {
+            // there are exactly the right number of compliant nominees and none have been blacklisted
+            nominees = SecurityCouncilNomineeElectionGovernorCountingUpgradeable.nominees(proposalId);
+        }
+    }
+
+    /// @dev    `GovernorUpgradeable` function to execute a proposal overridden to handle nominee elections.
+    ///         Can be called by anyone via `execute` after voting and blacklisting periods have ended.
+    ///         If the number of compliant nominees is > the target number of nominees, 
+    ///         we move on to the next phase by calling the SecurityCouncilMemberElectionGovernor.
+    ///         If the number of compliant nominees is == the target number of nominees,
+    ///         we execute the election result immediately by calling the SecurityCouncilManager.
+    ///         If the number of compliant nominees is < the target number of nominees,
+    ///         we randomly add some members from the current cohort to the list of nominees and then call the SecurityCouncilManager.
+    /// @param  proposalId The id of the proposal
+    function _execute(
+        uint256 proposalId,
+        address[] memory /* targets */,
+        uint256[] memory /* values */,
+        bytes[] memory /* calldatas */,
+        bytes32 /*descriptionHash*/
+    ) internal virtual override {
+        uint256 blacklistDeadline = proposalDeadline(proposalId) + foundationBlacklistDuration;
+        require(block.number > blacklistDeadline, "Proposal is still in the blacklist period");
+
+        uint256 blacklistedNomineeCount_ = blacklistedNomineeCount[proposalId];
+        uint256 compliantNomineeCount = nomineeCount(proposalId) - blacklistedNomineeCount_;
+
+        if (compliantNomineeCount > targetNomineeCount) {
+            // todo:
```

Comment:
if we're moving on to the next governor, we don't _necessarily_ need to read out the nominees and filter. instead, it's currently set up so that the `SecurityCouncilMemberElectionGovernor` calls `SecurityCouncilNomineeElectionGovernor.isCompliantNominee` so we avoid reading and writing to storage for each nominee at this step. 

currently `_determineCompliantNominees` returns an array of length equal to `targetNomineeCount`.

changing it such that `_determineCompliantNominees` returns an array of length `>= targetNomineeCount` that gets passed on to the next governor might make logic simpler, but could also be expensive if there are many nominees. I'm unsure which approach is better.